mypy/nodes.py

```diff
@@ -1946,6 +1946,27 @@ def __init__(self, names: 'SymbolTable', defn: ClassDef, module_name: str) -> No
             for vd in defn.type_vars:
                 self.type_vars.append(vd.name)
 
+    declared_metaclass = None  # type: Optional[mypy.types.Instance]
+    metaclass_type = None  # type: Optional[mypy.types.Instance]
+
+    def calculate_metaclass_type(self) -> 'Optional[mypy.types.Instance]':
+        if self.mro is None:
+            # XXX why does this happen?
```

Comment:
Turns out it happens during deserialization in incremental mode: the type of a direct base class may be None. (I should have looked into it long time ago. Sorry)